src/System.IO.FileSystem/tests/File/GetSetTimes.cs

```diff
@@ -1,118 +1,132 @@
-                            actual = File.GetLastWriteTime(fileName);
-                            actualUtc = File.GetLastWriteTimeUtc(fileName);
-                            break;
-                        default:
-                            throw new ArgumentException("Invalid time property type");
-                    }
+        [Fact]
+        public void EmptyPath_ThrowsArgumentException()
+        {
+            Assert.All(TimeFunctions(), (tuple) =>
+            {
+                Assert.Throws<ArgumentException>(() => tuple.Item1(string.Empty, DateTime.Today));
+                Assert.Throws<ArgumentException>(() => tuple.Item2(string.Empty));
+            });
+        }
 
-                    DateTime expected = dt.ToLocalTime();
-                    DateTime expectedUtc = dt.ToUniversalTime();
+        [Fact]
+        public void SettingUpdatesProperties()
+        {
+            FileInfo testFile = new FileInfo(GetTestFilePath());
+            testFile.Create().Dispose();
 
-                    if (dt.Kind == DateTimeKind.Unspecified)
-                    {
-                        if (setUtc)
-                        {
-                            expectedUtc = dt;
-                        }
-                        else
-                        {
-                            expected = dt;
-                        }
-                    }
+            Assert.All(TimeFunctions(), (tuple) =>
+            {
+                DateTime dt = new DateTime(2014, 12, 1, 12, 0, 0, tuple.Item3);
+                tuple.Item1(testFile.FullName, dt);
+                Assert.Equal(dt, tuple.Item2(testFile.FullName));
+            });
+        }
 
-                    Assert.Equal(expected, actual); //, "Local {0} should be correct for DateTimeKind.{1} when set with Set{0}{2}", timeProperty, kind, setUtc ? "Utc" : "");
-                    Assert.Equal(expectedUtc, actualUtc); //, "Universal {0} should be correct for DateTimeKind.{1} when set with Set{0}{2}", timeProperty, kind, setUtc ? "Utc" : "");
-                }
-            }
+        [OuterLoop]
+        [Fact]
+        public void CreationSetsAllTimes()
+        {
+            string path = GetTestFilePath();
+            long beforeTime = DateTime.Now.Ticks;
+            long utcBeforeTime = DateTime.UtcNow.Ticks;
+            Task.Delay(2000).Wait(); //buffer to resolve creation time
+
+            FileInfo testFile = new FileInfo(GetTestFilePath());
+            testFile.Create().Dispose();
+            Task.Delay(2000).Wait();
+
+            long afterTime = DateTime.Now.Ticks;
+            long utcAfterTime = DateTime.UtcNow.Ticks;
+
+            Assert.All(TimeFunctions(), (tuple) =>
+            {
+                Assert.InRange(tuple.Item2(testFile.FullName).Ticks, tuple.Item3 == DateTimeKind.Local ? beforeTime : utcBeforeTime, tuple.Item3 == DateTimeKind.Local ? afterTime : utcAfterTime);
+            });
         }
 
-        File.Delete(fileName);
+        [OuterLoop]
```

Comment:
My justification for OuterLoop was that it was using Task.Delay and was time-sensitive. Since I've replaced Task.Delay with AddSeconds(-n), that reasoning disappears. I will remove the OuterLoops.